sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -29,13 +38,40 @@ export interface LinkEntityOptions {
   audience?: string;
 }
 
+/**
+ * A simple grouping of the sender and receiver options. Only used
+ * with the ManagementClient today.
+ *
+ * @internal
+ * @ignore
+ */
+export interface RequestResponseLinkOptions {
+  senderOptions: SenderOptions;
+  receiverOptions: ReceiverOptions;
+  name?: string;
+}
+
+/**
+ * @internal
+ * @ignore
+ */
+type LinkOptionsT<
+  LinkT extends Receiver | AwaitableSender | RequestResponseLink
+> = LinkT extends Receiver
+  ? ReceiverOptionsWithSession
+  : LinkT extends AwaitableSender
+  ? AwaitableSenderOptions
+  : LinkT extends RequestResponseLink
+  ? RequestResponseLinkOptions
+  : never;
+
 /**
  * @internal
  * @ignore
  * Describes the base class for entities like MessageSender, MessageReceiver and Management client.
  * @class ClientEntity
```

Comment:
Nit: `@class` tag is not needed in TypeScript files. The class name is wrong as well. So, we can just remove it

```suggestion
```